README.md

```diff
@@ -2,19 +2,71 @@
 
 Let your software roar!
 
-## Architecture
-Tuesday concept:
+## Assumptions and Requirements
+Let's start with an analogy: the profiled program is like a mechanical gearbox with many cogs that each can make a sound. The sound a cog makes depends on 1) how fast it spins 2) how much power it transmits 3) the shape/material/kind of the cog.
+
+Acoustic profiler makes it fairly easy to
+1. implement probes for various types of cogs
+2. vary sound effects given to individual cogs
+3. hear cogs spinning in remote machine (host)
+
+This necessitates an interface somewhere between the probes and the sound synthesis so that they can be mix-and-matched. This interface also works over network to satisfy 3.
+
+## Interface
+
+On abstract level, the interface is just a stream of Events where event is
+
+```rust
+struct Event {
+    type: EventType
+    // only values between [0, 1]
+    quality: f32
+}
```

Comment:
I also slightly lean towards what Matej suggests. `quality` sounds nice and abstract, but I think we benefit from explicitness during the early iterations. As the project matures, we'll probably know better what level of abstraction we eventually want.

Reply:
Agreed on both counts :)

Alternatively, if there's an unbounded/unknown amount of possible event kinds, rather than an `enum` the right solution may be using dynamic dispatch and storing a boxed trait instead. This would increase complexity though, so it has to be warranted.

Reply:
> if each event knows what particular qualities it can have, then these may be more expressive as fields on the individual types:

Yeah, this looks better for the API layer. The layer that maps any event variant to any sound effect can then be internal to the server. We may actually find out that this layer can't be reasonably generic and we'll have have to implement an adapter for every `(event, effect)` pair individually.


I'll leave the readme as it is and merge this and I'll make sure to update it once we have actual code in place.